test/BUILD

```diff
@@ -1,6 +1,21 @@
 package(default_visibility = ["//visibility:public"])
 
-load("//bazel:envoy_build_system.bzl", "envoy_cc_test_library")
+load("//bazel:envoy_build_system.bzl", "envoy_cc_test", "envoy_cc_test_library")
+
+envoy_cc_test(
+    name = "example_configs_test",
+    srcs = [
+        "example_configs_test.cc",
+        "example_configs_test.h",
+    ],
+    data = ["//configs:example_configs"] + ["example_configs_test_setup.sh"],
```

Comment:
Did you disable sandboxing on the Bazel command-line?

When I patch in your diff and run with `bazel test //test:example_configs_test` I get:

```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
-----------------------------------------------------------------------------
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from ExampleConfigsTest
[ RUN      ] ExampleConfigsTest.All
sh: /usr/local/google/home/htuch/.cache/bazel/_bazel_htuch/89676793239ac96d94294e6c7a44597f/bazel-sandbox/5cc5c4f1-3666-42c7-a568-0209bdc4f93e-832/execroot/envoy/bazel-out/local-fastbuild/bin/test/example_configs_test.runfiles/envoy/test/example_configs_test_setup.sh: Permission denied
Failed /usr/local/google/home/htuch/.cache/bazel/_bazel_htuch/89676793239ac96d94294e6c7a44597f/bazel-sandbox/5cc5c4f1-3666-42c7-a568-0209bdc4f93e-832/execroot/envoy/bazel-out/local-fastbuild/bin/test/example_configs_test.runfiles/envoy/test/example_configs_test_setup.sh 
[2017-04-12 14:25:58.867][7][critical][assert] assert failure: false: test/test_common/environment.cc:117
external/bazel_tools/tools/test/test-setup.sh: line 159:     7 Aborted                 (core dumped) "${TEST_PATH}" "$@"
Use --strategy=TestRunner=standalone to disable sandboxing for the failing actions.
```
which seems to indicate Bazel test sandboxing is working as expected, so yeah, `${TEST_TMPDIR}` is the way to go.